mtcpclient/reporting.go

```diff
@@ -8,10 +8,13 @@ import (
 	"github.com/dachad/tcpgoon/tcpclient"
 )
 
-func collectConnectionsStatus(connectionsStatusRegistry GroupOfConnections, statusChannel <-chan tcpclient.Connection) {
+func collectConnectionsStatus(connectionsStatusRegistry *GroupOfConnections, statusChannel <-chan tcpclient.Connection) {
 	for {
 		newConnectionStatusReported := <-statusChannel
-		connectionsStatusRegistry[newConnectionStatusReported.ID] = newConnectionStatusReported
+		connectionsStatusRegistry.connections[newConnectionStatusReported.ID] = newConnectionStatusReported
+		if len(connectionsStatusRegistry.getFilteredListByStatus([]tcpclient.ConnectionStatus{tcpclient.ConnectionEstablished})) > connectionsStatusRegistry.metrics.maxConcurrentEstalished {
+			connectionsStatusRegistry.metrics.maxConcurrentEstalished = len(connectionsStatusRegistry.getFilteredListByStatus([]tcpclient.ConnectionStatus{tcpclient.ConnectionEstablished}))
```

Comment:
and you count again / new full scan, and this goroutine is going to be called as many connections * number of status changes, that can be considerable